src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -90,53 +91,59 @@ internal ReflectTypeDescriptionProvider()
         {
         }
 
-        private static Hashtable InitIntrinsicTypeConverters()
-        {
-            // Add the intrinsics
-            return new Hashtable
-            {
-                [typeof(bool)] = typeof(BooleanConverter),
-                [typeof(byte)] = typeof(ByteConverter),
-                [typeof(SByte)] = typeof(SByteConverter),
-                [typeof(char)] = typeof(CharConverter),
-                [typeof(double)] = typeof(DoubleConverter),
-                [typeof(string)] = typeof(StringConverter),
-                [typeof(int)] = typeof(Int32Converter),
-                [typeof(short)] = typeof(Int16Converter),
-                [typeof(long)] = typeof(Int64Converter),
-                [typeof(float)] = typeof(SingleConverter),
-                [typeof(UInt16)] = typeof(UInt16Converter),
-                [typeof(UInt32)] = typeof(UInt32Converter),
-                [typeof(UInt64)] = typeof(UInt64Converter),
-                [typeof(object)] = typeof(TypeConverter),
-                [typeof(void)] = typeof(TypeConverter),
-                [typeof(DateTime)] = typeof(DateTimeConverter),
-                [typeof(DateTimeOffset)] = typeof(DateTimeOffsetConverter),
-                [typeof(Decimal)] = typeof(DecimalConverter),
-                [typeof(TimeSpan)] = typeof(TimeSpanConverter),
-                [typeof(Guid)] = typeof(GuidConverter),
-                [typeof(Uri)] = typeof(UriTypeConverter),
-                [typeof(Color)] = typeof(ColorConverter),
-                [typeof(Point)] = typeof(PointConverter),
-                [typeof(Rectangle)] = typeof(RectangleConverter),
-                [typeof(Size)] = typeof(SizeConverter),
-                [typeof(SizeF)] = typeof(SizeFConverter),
-                // Special cases for things that are not bound to a specific type
-                //
-                [typeof(Array)] = typeof(ArrayConverter),
-                [typeof(ICollection)] = typeof(CollectionConverter),
-                [typeof(Enum)] = typeof(EnumConverter),
-                [s_intrinsicNullableKey] = typeof(NullableConverter),
-            };
-        }
+        private static Hashtable EditorTables => LazyInitializer.EnsureInitialized(ref s_editorTables, () => new Hashtable(4));
```

Comment:
> The ask in current task is to eliminate memory barriers which are a consequence of volatile fields.

That's not the goal.  The goal is to ensure that lazy initialization is done correctly, safely, and consistently, while minimizing the code required to do so.

> But then EnsureInitialized appears to rely on similar strategy involving volatile reads
  
Any thread-safe lazy initialization strategy is going to rely on volatile reads.  That ```Lazy<T>``` doesn't appear to be reading ```_boxed``` via a volatile read technically looks like a bug to me.  Now, due to the semantics enforced by the runtime, it's likely going to be made volatile by the runtime anyway, such that the difference wouldn't be noticeable, and even then on x32/64 the hardware enforces a strong memory model such that it further wouldn't be noticeable, but it's still good to do the right thing in the code.